File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int ans = Integer.MAX_VALUE;
        int i=0, j = nums.length - 1;
        while(i <= j){
            int mid = i + (j - i)/2;
            if(nums[i] <= nums[mid]){
                ans = Math.min(ans, nums[i]);
                i = mid + 1;
            } else{
                ans = Math.min(ans, nums[mid]);
                j = mid - 1;
            }
        }
        return ans;
    }
}

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int n = nums.size();
        int i = 0, j = n-1;
        while(i < j) {
            int mid = i + (j - i)/2;
            if(nums[mid] > nums[j]){
                i = mid + 1;
            } else{
                j = mid; // mid -1 is not taken because there can be a chance that mid is the smallest one , so to track that and even also the while loop condition is i<j so if i == j then also the loop will break and the answer will become nums[i] as it refers to the left most point where the value is smaller one
            }
        }
        return nums[i];
    }
};